### PR TITLE
Added pycatchmod.io and click module

### PR DESCRIPTION
**Pull request by snorfalorpagus**
This PR bundles a few changes together:

- Added a command line interface using click, "pycatchmod"
- Added `io` submodule to read excel data (parameters, inputs, results)
- Added a `run_catchmod` function for convenience

### CODE REVIEW

**Comment by snorfalorpagus** [comment]
I think there are still some issues with the way the Excel model stores some of the parameters, but that's left for another issue/PR.

File: pycatchmod/__main__.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    from pycatchmod.io.json import catchment_from_json
+    from pycatchmod import run_catchmod
+
+    # load rainfall and pet data
+    rainfall = pandas_read(rainfall)
+    pet = pandas_read(pet)
+    num_scenarios = rainfall.shape[1]
+
+    # load catchmod model
+    catchment = catchment_from_json(parameters, n=num_scenarios)
```
Comment (jetuk) [inline pycatchmod/__main__.py]: Could this be from excel too?

Comment (snorfalorpagus) [inline pycatchmod/__main__.py]: I envisioned the using the dump command to create the JSON from Excel explicitly. The next step is to add a control file that contains all of the information, i.e. which parameters.json and which rainfall/PET to use.

**Comment by snorfalorpagus** [comment]
There aren't any tests for the Excel parts because we'd have to upload one of the Excel models, which I don't think we can share. I guess we could create something that had the same structure?

**Comment by jetuk** [comment]
I guess it would then be very clear what Excel layout the code supports?

**Comment by snorfalorpagus** [comment]
I've added some tests for the Excel read functions that use a dummy "thames.xls" file. This file includes the minimum amount of data for the functions to work.

I think this is ready for merge now. There are some outstanding issues related to the different ways parameters are stored in Excel / pycatchmod and I'd also like to add a flag to the CLI to not automatically add leap days, but those are for another PR.

**Pull request merged**